src/Client.js

```diff
@@ -384,17 +383,37 @@ Client.prototype._handleRequestResult = function(response, result, options) {
   errors.FaunaHTTPError.raiseForStatusCode(result)
 }
 
-function getHttp2SessionIdleTime() {
-  var fromEnv = util.getEnvVariable('FAUNADB_HTTP2_SESSION_IDLE_TIME')
-  var parsed =
-    // Allow either "Infinity" or parsable integer string.
-    fromEnv === 'Infinity' ? Infinity : parseInt(fromEnv, 10)
-  var useEnvVar = !isNaN(parsed)
+function getHttp2SessionIdleTime(configuredIdleTime) {
+  const maxIdleTime = 5000
+  const defaultIdleTime = 500
+  const envIdleTime = util.getEnvVariable('FAUNADB_HTTP2_SESSION_IDLE_TIME')
 
-  return {
-    shouldOverride: useEnvVar,
-    value: useEnvVar ? parsed : 500,
+  var value = defaultIdleTime
+  // attemp to set the idle time to the env value and then the configured value
+  const values = [envIdleTime, configuredIdleTime]
+  for (let i = 0; i <= values.length; i++) {
+    const rawValue = values[i]
+    const parsedValue = parseInt(rawValue, 10)
+    const isNegative = parsedValue < 0
+    const isInfinity = rawValue === 'Infinity'
+    const isGreaterThanMax = parsedValue > maxIdleTime || isInfinity
+    // if we didn't get infinity or a positive integer move to the next value
+    if (isNegative) continue
+    if (!isInfinity && !parsedValue) continue
+    // if we did get something valid constrain it to the ceiling
+    value = parsedValue
+    if (isGreaterThanMax) {
+      value = maxIdleTime
+      console.warn(
+        `The value set for http2SessionIdleTime exceeds the maximum value of 
+        ${maxIdleTime} milliseconds. It will be set to ${maxIdleTime} milliseconds instead.`
+      )
```

Comment:
i think we should remove logging like this.

Since this is a library we don't want to spam the user's context, and I don't see anywhere else in the driver we're doing this.

If we want to build in some sort of notification system we should probably think through a strategy before committing.

Let's make up for this gap by clarifying the documentation further.

```suggestion
```